### ISSUE TITLE
[Enhancement] Add MD5 or SHA256 or SHA512 checksum/hash to AppImageKit releases

### ISSUE THREAD

**Issue by KurtPfeifle** [AppImage/AppImageKit#595]
#### TL;DR: supply checksum / hashes alongside the download links for binaries
----

It's not the first time that developers of this project claim a user reporting issues didn't use the latest versions of AppImageKit tools when the issue reporter in question in fact had just downloaded or updated the respective tools, or vice versa -- the developer had himself not checked with the latest version at hand.

[Example 1:](https://github.com/AppImage/AppImageKit/issues/593#issuecomment-352807990) *" I have tested it with the latest continuous build, which I downloaded with AppImageUpdate as well. **Your build is just not the one I have**."*

[Example 2:](https://github.com/AppImage/AppImageKit/issues/545#issuecomment-346704727) *"[...] double-check if you was testing the right version?"*

It's also not the first time when I suspect that sometimes, somehow downloads from the Github release page may not deliver what they claim they do, or may even lead to subtly corrupted downloads. I've no idea how they use external content delivery networks (Akamai, Amazon, Cloudflare & Co.), and how caching, binary deltas and other techniques may interfere and make things "work" in unexpected ways...

Also, using the CLI parameter *`--appimage-version`* is not the best way to compare binaries, because:

1. It gives the version of the AppImageKit used to build the tool, not the version of the tool itself.
1. In some cases it still returns a relatively meaningless generic string like '`continuous`* (examples: recent versions of [appimagetool-x86_64.AppImage](https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage).

The (optional) version string used in an original filename may have been removed when re-naming the file manually, or by some automatic mechanism when updating the tool.

Any of the MD5/SHA256/SHA512 hashes would give a better way to resolve ambiguities about different versions in circulations amongst users and developers then the current *`--appimage-version`* method. It would additionally provide a means to verify downloads and uncover corrupted ones if there is a doubt.

**Comment by probonopd** [comment]
Maybe we can use the hashing code that we have already written and that is being used for GPG-signing AppImages. It is already running when the user uses the `-g` option. Maybe we should run it when the user is not using that option, too, and embed the resulting hash somewhere.

**Comment by KurtPfeifle** [comment]
> *"Maybe we can use the hashing code that we have already written and that is being used for GPG-signing AppImages."*

Huh? -- I'm not asking to embed the hash inside the AppImage. Just create the hash after the AppImage is built and publish it alongside the main product(s) on the GitHub release page. Sorry if I didn't express myself clearly enough.

**Comment by probonopd** [comment]
In fact we are already publishing the SHA-1 checksum as part of the zsync file:

```
me@host:~$ strings Android_File_Transfer-b44ff25-x86_64.AppImage.zsync | grep ^SHA-1 | head -n 1 | cut -d " " -f 2
2902ba60b8bb376425f6180db4c819b0d9a24532
me@host:~$ sha1sum Android_File_Transfer-b44ff25-x86_64.AppImage
2902ba60b8bb376425f6180db4c819b0d9a24532  Android_File_Transfer-b44ff25-x86_64.AppImage
```

**Comment by KurtPfeifle** [comment]
> *"In fact we are already publishing the SHA-1 checksum as part of the zsync file:"*

Interesting...

But... (hehe ;-) ) ...I would't dare to call this *"publishing"*. If you had said *"we are already **hiding** the SHA-1 checksum as part of the zsync file"* I'd subcribe to this statement.
